src/test/java/uk/gov/hmcts/reform/translate/service/DictionaryServiceTest.java

```diff
@@ -218,12 +241,180 @@ void shouldUpdateADictionaryForUserWithoutManageTranslationsRole() {
+            dictionaryService.putDictionaryRoleCheck(CLIENTS2S_TOKEN);
+            dictionaryService.putDictionary(dictionaryRequest);
+
+            verify(dictionaryRepository, times(3)).findByEnglishPhrase(any());
+            verify(securityUtils, times(1)).hasRole(anyString());
+            verify(dictionaryMapper, times(3)).modelToEntityWithTranslationUploadEntity(any(), any());
+            verify(dictionaryRepository, times(3)).save(any());
+
+        }
+
+        @Test
+        void shouldPutDictionaryRoleCheckForAValidUserWithLoadTranslationRole() {
+            final Dictionary dictionaryRequest = getDictionaryRequestWithoutABody(1, 2);
+            final DictionaryEntity dictionaryEntity =
+                createDictionaryEntity("english_1", "translated_1");
+            given(applicationParams.getPutDictionaryS2sServicesBypassRoleAuthCheck())
+                .willReturn(Arrays.asList(DEFINITION_STORE));
+            given(securityUtils.hasAnyOfTheseRoles(anyList())).willReturn(true);
+            given(securityUtils.getServiceNameFromS2SToken(CLIENTS2S_TOKEN)).willReturn(XUI);
+
+            given(dictionaryRepository.findByEnglishPhrase(any())).willReturn(Optional.of(dictionaryEntity));
+            given(securityUtils.getUserInfo()).willReturn(getUserInfoWithManageTranslationsRole());
+            given(securityUtils.hasRole(anyString())).willReturn(true);
+            dictionaryService.putDictionaryRoleCheck(CLIENTS2S_TOKEN);
+            dictionaryService.putDictionary(dictionaryRequest);
+
+            verify(dictionaryRepository, times(1)).findByEnglishPhrase(any());
+            verify(securityUtils, times(1)).hasRole(anyString());
+            verify(dictionaryRepository, times(1)).save(any());
+        }
+
+        @Test
+        void shouldPutDictionaryRoleCheckForAValidDefinitionStore() {
+
+            final Dictionary dictionaryRequest = getDictionaryRequestWithoutABody(1, 2);
+            final DictionaryEntity dictionaryEntity =
+                createDictionaryEntity("english_1", "translated_1");
+            given(applicationParams.getPutDictionaryS2sServicesBypassRoleAuthCheck())
+                .willReturn(Arrays.asList(DEFINITION_STORE));
+            given(securityUtils.getServiceNameFromS2SToken(CLIENTS2S_TOKEN)).willReturn(DEFINITION_STORE);
+
+            given(dictionaryRepository.findByEnglishPhrase(any())).willReturn(Optional.of(dictionaryEntity));
+            given(securityUtils.getUserInfo()).willReturn(getUserInfoWithManageTranslationsRole());
+            given(securityUtils.hasRole(anyString())).willReturn(true);
+            dictionaryService.putDictionaryRoleCheck(CLIENTS2S_TOKEN);
+            dictionaryService.putDictionary(dictionaryRequest);
+
+            verify(dictionaryRepository, times(1)).findByEnglishPhrase(any());
+            verify(securityUtils, times(1)).hasRole(anyString());
+            verify(dictionaryRepository, times(1)).save(any());
+
+        }
```

Comment:
These tests should just be checking behaviour of `dictionaryService.putDictionaryRoleCheck` as behavior of `dictionaryService.putDictionary(dictionaryRequest)` is checked elsewhere.

NB: As for these happy path tests `dictionaryService.putDictionaryRoleCheck` returns no values you can use `assertDoesNotThrow` to apply some test on the call.

Reply:
Resolved in c655d3d72b8f1560acb96d062f9650d0b82728af